runtime/runtime/setup.go

```diff
@@ -107,40 +101,38 @@ func (srv *Server) handler(w http.ResponseWriter, req *http.Request) {
 	if h := req.Header.Get("X-Encore-Auth"); h != "" {
 		if ok, err := srv.checkAuth(req, h); err == nil && ok {
 			// Successfully authenticated
+			req = req.WithContext(withEncoreAuthentication(req.Context()))
 			r = srv.private
 		} else if err != nil {
-			http.Error(w, "could not authenticate request", http.StatusBadGateway)
+			errs.HTTPError(w, errs.B().Code(errs.Internal).Msg("could not authenticate request").Err())
 			return
 		} else {
-			http.Error(w, "invalid request signature", http.StatusUnauthorized)
+			errs.HTTPError(w, errs.B().Code(errs.Unauthenticated).Msg("invalid request signature").Err())
 			return
 		}
 	}
 
-	// We use EscapedPath rather than `req.URL.Path` because if the path contains an encoded
-	// forward slash as %2F we don't want the router to treat that as a segment split.
-	//
-	// i.e. `/foo%2Fbar/baz` should be routed to `/:a/*b` as a = "foo/bar", b = "baz"
-	// where as if we use req.URL.Path we would get a = "foo", b = "bar/baz` which is incorrect.
-	path := req.URL.EscapedPath()
+	// Switch to the Encore internal router if we are on a Encore internal path
+	// Historically we used a prefix of `/__encore` but this is now deprecated in favour of a well-known `encore` path
+	// But for now we support both
+	if strings.HasPrefix(path, "/__encore/") || strings.HasPrefix(path, "/.well-known/encore/") {
```

Comment:
Not sold on `.well-known/encore` over `__encore`. It seems primarily used for informational things and not for APIs. What's the reason for moving it?

Reply:
They're not just for informational use, but also "well-known services". As such I've seen a couple pieces of software put their health endpoint inside the well-known directory ([for example, Apollo](https://www.apollographql.com/docs/apollo-server/monitoring/health-checks/#http-level-health-checks)).

The reasoning is this feels more inline with not stomping over an application URL space, and putting it in a clearly well known location with a clear identifier for what we're talking about.